apps/test-suite/tests/FileSystemNext.ts

```diff
@@ -0,0 +1,95 @@
+'use strict';
+import * as FS from 'expo-file-system';
+import { File, Directory } from 'expo-file-system/next';
+import { Platform } from 'react-native';
+
+export const name = 'FileSystem@next';
+
+export async function test({ describe, expect, it, ...t }) {
+  const testDirectory = FS.documentDirectory + 'tests/';
+  t.beforeEach(async () => {
+    try {
+      await FS.makeDirectoryAsync(testDirectory);
+    } catch {}
+  });
+  t.afterEach(async () => {
+    try {
+      await FS.deleteAsync(testDirectory);
+    } catch {}
+  });
+
+  describe('FileSystem (Next)', () => {
+    it('Creates a lazy file reference', async () => {
+      const file = new File('file:///path/to/file');
+      // The path is normalized by the OS, with both being valid urls (https://stackoverflow.com/a/44725349)
+      if (Platform.OS === 'ios') {
+        expect(file.path).toBe('file:///path/to/file');
+      } else if (Platform.OS === 'android') {
+        expect(file.path).toBe('file:/path/to/file');
+      }
+    });
+
+    it('Allows changing the path property', async () => {
+      const file = new File('file:///path/to/file');
+      file.path = 'file:///new/path';
```

Comment:
I don't think we need that, but I'm open to discussing that. In my opinion, the `File` should be immutable. If you what to mutate the path based on the previous one, I would rather have API like this:
```
const dir = new File('file:/app/folder/1')
const file = new File(dir, 'avatar.jpg')
```

Reply:
I think moving a file should update an underlying path reference, so Files are mutable anyways.
Do you agree?

```
const file = new File('file:/app/folder/1')
file.copy("file:/app/folder/2")
file.path == "file:/app/folder/2"
```

If we make files mutable by some operations, I don't see a good reason not to allow manually mutating the path, but also no big usecases for it.

Reply:
I think we already discussed this – copy operation shouldn't mutate the path, instead it could return a new file.
It probably makes sense to mutate the path on move, but we can mutate this only on the native side. In other words, imho the path should be mutable, but read-only on the JS side.

Reply:
I believe that when a file is moved or copied, it should create a new file object rather than changing the original file. When a file is moved, the old file should refer to the previous path where the file was located. I don't think the path should always lead to a valid file because we also want to be able to check if the file exists. I would consider the 'File' object as a pointer that may lead to something that hasn't been created yet.

Reply:
I agree on what a `File` is in our implementation, but I think even if you have a pointer to a location, and you call move on an instance of that pointer, I'd expect it to point to that new location if that operation succeeded – i.e. I weakly disagree on `When a file is moved, the old file should refer to the previous path where the file was located.`